mmedit/models/common/ensemble.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+import torch.nn as nn
+
+
+class SpatialTemporalEnsemble(nn.Module):
+    """ Apply spatial and temporal ensemble and compute outputs
+
+    Args:
+        is_temporal_ensemble (bool, optional): Whether to apply ensemble
+            temporally. If True, the sequence will also be flipped temporally.
+            If the input is an image, this argument must be set to False.
+            Default: False.
+
+    Returns:
+        Tensor: Model outputs given the ensembles.
+    """
+
+    def __init__(self, is_temporal_ensemble=False):
+
+        super().__init__()
+
+        self.is_temporal_ensemble = is_temporal_ensemble
+
+    def _transform(self, imgs, mode):
+        is_single_image = False
+        if imgs.ndim == 4:
+            if self.is_temporal_ensemble:
+                raise ValueError('"is_temporal_ensemble" must be False if '
+                                 'the input is an image.')
+            is_single_image = True
+            imgs = imgs.unsqueeze(1)
+
+        if mode == 'vertical':
+            imgs = imgs.flip(4).clone()
```

Comment:
Only support video
or
Support video and single image

how about 
`flip(-1)`
`transpose(-2, -1)`

Reply:
If the input is an image (n, c, h, w), it is unsqueezed to (n, 1, c, h, w). So it should be fine.